File: lang/go/test/fun/vac/algorithms/sorting/MergeSort/MergeSortTest.go
package main

import (
    "fun/vac/algorithms/sorting/MergeSort"
    "fun/vac/util/SequenceBuilder"
    "fun/vac/util/Sequences"
    "fun/vac/util/TestRunner"
)

func testMergeSort() bool {
    size := 32768

    arr := make([]int, size)
    SequenceBuilder.PackRandom(arr)

    checksum := Sequences.ParityChecksum(arr)

    MergeSort.Sort(arr)

    if Sequences.ParityChecksum(arr) != checksum {
        return false
    }

    return Sequences.IsSorted(arr)
}

func main() {
    TestRunner.ParseTest(testMergeSort())
}

File: lang/c/test/vac.fun/algorithms/sorting/MergeSortTest.c
#include <vac.fun/algorithms/sorting/MergeSort.h>
#include <vac.fun/util/SequenceBuilder.h>
#include <vac.fun/util/Sequences.h>
#include <vac.fun/util/TestRunner.h>

bool testMergeSort(void) {
    size_t size = 32768;

    int arr[size];
    SequenceBuilder_packRandom(arr, size);

    int checksum = Sequences_parityChecksum(arr, size);

    MergeSort_sort(arr, size);

    if (Sequences_parityChecksum(arr, size) != checksum) {
        return false;
    }

    return Sequences_isSorted(arr, size);
}

int main(void) {
    TestRunner_parseTest(testMergeSort());
}
